tests/unit/test_clusters.py

```diff
@@ -0,0 +1,206 @@
+import ast
+import json
+import responses
+import unittest
+from unittest.mock import patch
+
+from ansible.module_utils import basic
+from ansible.module_utils.common.text.converters import to_bytes
+
+from plugins.modules import clusters
+
+API_VERSION = "v2"
+API_URL = f"https://api.openshift.com/api/assisted-install/{API_VERSION}"
+
+TEST_JSON = {"test": ["test1", "test2"]}
+
+
+def set_module_args(args):
+    """prepare arguments so that they will be picked up during module creation"""
+    args = json.dumps({"ANSIBLE_MODULE_ARGS": args})
+    basic._ANSIBLE_ARGS = to_bytes(args)
+
+
+class AnsibleExitJson(Exception):
+    """Exception class to be raised by module.exit_json and caught by the test case"""
+
+    pass
+
+
+class AnsibleFailJson(Exception):
+    """Exception class to be raised by module.fail_json and caught by the test case"""
+
+    pass
+
+
+def exit_json(*args, **kwargs):
+    """function to patch over exit_json; package return data into an exception"""
+    if "changed" not in kwargs:
+        kwargs["changed"] = False
+    raise AnsibleExitJson(kwargs)
+
+
+def fail_json(*args, **kwargs):
+    """function to patch over fail_json; package return data into an exception"""
+    kwargs["failed"] = True
+    raise AnsibleFailJson(kwargs)
+
+
+def get_bin_path(self, arg, required=False):
+    """Mock AnsibleModule.get_bin_path"""
+    if arg.endswith("clusters"):
+        return "/usr/bin/clusters"
+    else:
+        if required:
+            fail_json(msg="%r not found !" % arg)
+
+
+class TestClusters(unittest.TestCase):
+
+    def setUp(self):
+        self.mock_module_helper = patch.multiple(
+            basic.AnsibleModule,
+            exit_json=exit_json,
+            fail_json=fail_json,
+            get_bin_path=get_bin_path,
+        )
+        self.mock_module_helper.start()
+        self.addCleanup(self.mock_module_helper.stop)
+
+    @responses.activate
```

Comment:
Where would the equivalent of responses.add() which sets up the mock response be found? Thanks

Reply:
In this example it would be https://github.com/vjayaramrh/assistedinstaller/blob/3f8a79ef7c4656f27ff9b5eb07c203edacdbed1b/tests/unit/test_clusters.py#L76

responses.get sets a mock get response with the given URL, json, status, etc. You can do the same for post, delete, put, etc.